duckduckgo/iOS
# Move feature flagging + internal user management to BSK.PrivacyConfig

Task/Issue URL:
https://app.asana.com/0/72649045549333/1203676877763836
Tech Design URL:
https://app.asana.com/0/481882893211075/1203938569856702
https://app.asana.com/0/481882893211075/1203984214904257
CC:

**Description**:

This PR in combination with [it's BSK equivalent](https://github.com/duckduckgo/BrowserServicesKit/pull/258) adds subfeatures support to the Privacy Config component, adds support for an additional state, internal, and moves the InternalUserDecider from iOS into BSK to be used by the AppPrivacyConfiguration to provide the correct state for internal users.

It also adds support for subfeatures to the autofill project, initially to hide in development features for non internal users, but eventually with the intention of allowing "kill-switch" type functionality so parts of autofill can be turned off remotely in the case of problems in production.

I'm mainly opening this as a draft to get some initial eyes on it before opening a final PR. It needs more testing (automated and manual). I'm going to be separating the subfeature part out in parallel as this is more pressing for the Autofill project.

It is currently pointing the BSK package to [this branch](https://github.com/duckduckgo/BrowserServicesKit/pull/258), for testing purposes, but that part will be reverted when the BSK changes are merged.



**Steps to test this PR**:
1. Check that any existing features configured using the privacy-config still work as expected
2. Check that email protection is still enabled
3. Check that credentials autofill is enabled if you're authenticated as an internal user, but not if you are not.



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
